pkg/actuators/machine/reconciler.go

```diff
@@ -287,7 +295,7 @@ func (r *Reconciler) exists() (bool, error) {
 		return false, err
 	}
 
-	if len(existingInstances) == 0 {
+	if len(existingInstances) == 0 || r.checkIfInstanceTerminated(existingInstances) {
 		if r.machine.Spec.ProviderID != nil && *r.machine.Spec.ProviderID != "" && len(r.machine.Status.Addresses) == 0 && (r.machine.Status.LastUpdated == nil || r.machine.Status.LastUpdated.Add(requeueAfterSeconds*time.Second).After(time.Now())) {
 			klog.Infof("%s: Possible eventual-consistency discrepancy; returning an error to requeue", r.machine.Name)
 			return false, &machinecontroller.RequeueAfterError{RequeueAfter: requeueAfterSeconds * time.Second}
```

Comment:
This might not be what we want. I think there's probably a case where the instance is terminated, but the provider ID is set, without the addresses in place, which makes me think we can probably hit this eventual consistency issue.

Given the last updated check this is probably ok since we wait circa 20 seconds before giving up, but, I'd like to see this tested if we can

Reply:
Thanks for your review Joel, from our testing we could see that it works how we hope. Will look into unit tests next